mkContainer/mkContainer.nix

```diff
@@ -8,72 +8,152 @@
   # the system to build for
   system,
   containerSystem,
+  environment ? builtins.storePath environmentOutPath,
+  nixpkgsFlake ? builtins.getFlake nixpkgsFlakeRef,
+  pkgs ? nixpkgsFlake.legacyPackages.${system},
+  containerPkgs ? nixpkgsFlake.legacyPackages.${containerSystem},
   containerName ? "flox-env-container",
   containerTag ? null,
   containerCreated ? "now",
-  containerConfig ? null,
+  containerConfigJSON ? "{}",
 }:
 let
-  environment = builtins.storePath environmentOutPath;
-  nixpkgsFlake = builtins.getFlake nixpkgsFlakeRef;
-  pkgs = nixpkgsFlake.legacyPackages.${system};
-  containerPkgs = nixpkgsFlake.legacyPackages.${containerSystem};
-  lib = pkgs.lib;
-  lowPriority =
-    pkg:
-    pkg.overrideAttrs (
-      old:
-      old
-      // {
-        meta = (old.meta or { }) // {
-          priority = 10000;
-        };
-      }
-    );
+  inherit (builtins)
+    fromJSON
+    listToAttrs
+    toString
+    elemAt
+    tryEval
+    match
+    length
+    ;
+  inherit (pkgs.lib)
+    nameValuePair
+    mapAttrsToList
+    optionalString
+    optionalAttrs
+    optionals
+    toIntBase10
+    splitString
+    last
```

Comment:
nit: looks like some of these might be stale? I don't see e.g. `last` used anywhere

Reply:
Yup.